ui/app/controllers/topology.js

```diff
@@ -1,23 +1,152 @@
   maximumFractionDigits: 2,
 });
 
 @classic
-export default class TopologyControllers extends Controller {
+export default class TopologyControllers extends Controller.extend(Searchable) {
   @service userSettings;
 
+  queryParams = [
+    {
+      searchTerm: 'search',
+    },
+    {
+      qpState: 'status',
+    },
+    {
+      qpVersion: 'version',
+    },
+    {
+      qpClass: 'class',
+    },
+    {
+      qpDatacenter: 'dc',
+    },
+  ];
+
+  @tracked searchTerm = '';
+
+  setFacetQueryParam(queryParam, selection) {
+    this.set(queryParam, serialize(selection));
+  }
+
+  @selection('qpState') selectionState;
+  @selection('qpClass') selectionClass;
+  @selection('qpDatacenter') selectionDatacenter;
+  @selection('qpVersion') selectionVersion;
+
+  @computed
+  get optionsState() {
+    return [
+      { key: 'initializing', label: 'Initializing' },
+      { key: 'ready', label: 'Ready' },
+      { key: 'down', label: 'Down' },
+      { key: 'ineligible', label: 'Ineligible' },
+      { key: 'draining', label: 'Draining' },
+      { key: 'disconnected', label: 'Disconnected' },
+    ];
+  }
+
+  @computed('model.nodes', 'nodes.[]', 'selectionClass')
+  get optionsClass() {
+    const classes = Array.from(new Set(this.model.nodes.mapBy('nodeClass')))
+      .compact()
+      .without('');
+
+    // Remove any invalid node classes from the query param/selection
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpClass',
+        serialize(intersection(classes, this.selectionClass))
+      );
+    });
+
+    return classes.sort().map((dc) => ({ key: dc, label: dc }));
+  }
+
+  @computed('model.nodes', 'nodes.[]', 'selectionDatacenter')
+  get optionsDatacenter() {
+    const datacenters = Array.from(
+      new Set(this.model.nodes.mapBy('datacenter'))
+    ).compact();
+
+    // Remove any invalid datacenters from the query param/selection
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpDatacenter',
+        serialize(intersection(datacenters, this.selectionDatacenter))
+      );
+    });
+
+    return datacenters.sort().map((dc) => ({ key: dc, label: dc }));
+  }
+
+  @computed('model.nodes', 'nodes.[]', 'selectionVersion')
+  get optionsVersion() {
+    const versions = Array.from(
+      new Set(this.model.nodes.mapBy('version'))
+    ).compact();
+
+    // Remove any invalid versions from the query param/selection
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpVersion',
+        serialize(intersection(versions, this.selectionVersion))
+      );
+    });
+
+    return versions.sort().map((v) => ({ key: v, label: v }));
+  }
+
   @alias('userSettings.showTopoVizPollingNotice') showPollingNotice;
 
-  @tracked filteredNodes = null;
+  @tracked pre09Nodes = null;
```

Comment:
Because we want to be able to filter nodes now, I've renamed what was `filteredNodes` to `pre09Nodes`. This modifies the code added from https://github.com/hashicorp/nomad/pull/9733 but I think in a healthy way. cc @DingoEatingFuzz